cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,185 @@
+package functions;
+
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import io.opencensus.common.Scope;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceConfiguration;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.samplers.Samplers;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.Properties;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.sql.DataSource;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+
+public class Main implements HttpFunction {
+
+  // TODO: replace these values before deploying
+  private static final String PROJECT_ID = "";
+  private static final String CONNECTION_NAME = "";
+  private static final String DB_NAME = "";
+  private static final String DB_USER = "";
+  private static final String DB_PASSWORD = "";
+
+  private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+  private static final Tracer TRACER = Tracing.getTracer();
+
+  static {
+    try {
+      // Set up StackDriver
+      StackdriverTraceExporter.createAndRegister(
+          StackdriverTraceConfiguration.builder()
+              .setProjectId(PROJECT_ID)
+              .build());
+
+    } catch (java.io.IOException e) {
+      LOGGER.log(Level.WARNING, "Warning: Could not set up Stackdriver Trace", e);
+    }
+
+  }
```

Comment:
I think you might want to use an initalization-on-demand holder here: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/functions/concepts/lazy-fields/src/main/java/functions/LazyFields.java#L35-L48

Reply:
done.